vf_convenience/cl_menu.lua

```diff
@@ -0,0 +1,109 @@
+CurrentSlot = 10
+gxt = "SNK_MNU"
+
+
+RegisterNetEvent('vf_convenience:update')
+AddEventHandler('vf_convenience:update', function()
+  if DoesStoreMenuExist then
+    _generalStorePool:Clear()
+    _generalStorePool:Remove()
+
+    -- Create the store menu
+    _generalStorePool = NativeUI.CreatePool()
+    storeMenu = NativeUI.CreateMenu("", "~s~" .. GetLabelText("SNK_ITEM"), "", "", MenuType, MenuType)
+    _generalStorePool:Add(storeMenu)
+
+    _generalStorePool:ControlDisablingEnabled(true)
+    _generalStorePool:MouseControlsEnabled(false)
+
+    Wait(10)
+
+    CreateStoreMenu(storeMenu)
+    _generalStorePool:RefreshIndex()
+    storeMenu:Visible(not storeMenu:Visible())
+    DoesStoreMenuExist = true
+  end
+end)
+
+Citizen.CreateThread(function()
+  while not NetworkIsGameInProgress() and not IsPlayerPlaying(PlayerId()) do
+    Citizen.Wait(0)
+  end
+
+  while HasAdditionalTextLoaded(CurrentSlot) and not HasThisAdditionalTextLoaded(gxt, CurrentSlot) do
+    Citizen.Wait(0)
+    CurrentSlot = CurrentSlot + 1
+  end
+
+  if not HasThisAdditionalTextLoaded(gxt, CurrentSlot) then
+    ClearAdditionalText(CurrentSlot, true)
+    RequestAdditionalText(gxt, CurrentSlot)
+    while not HasThisAdditionalTextLoaded(gxt, CurrentSlot) do
+      Citizen.Wait(0)
+    end
+  end
+
+  Citizen.Wait(1800)
+end)
+
+function CreateStoreMenu(menu)  
+  inventoryItems = exports.vf_base:GetInventory()
+  ConvenienceStore = {
+    {name = "SNK_ITEM1", description = "SNK_ITEM1_D", price = 10},
+    {name = "SNK_ITEM2", description = "SNK_ITEM2_D", price = 20},
+    {name = "SNK_ITEM3", description = "SNK_ITEM3_D", price = 10},
+    {name = "SNK_ITEM4", description = "SNK_ITEM4_D", price = 30},
+    {name = "SNK_ITEM5", description = "SNK_ITEM5_D", price = 50},
+    {name = "SNK_ITEM6", description = "SNK_ITEM6_D", price = 50} 
+  }
```

Comment:
Maybe move these to config.lua also so users can easily customize them.

Reply:
if i remember correctly this is using labels so adding custom product names would probably return a a NULL response.